<html xmlns:v="urn:schemas-microsoft-com:vml"xmlns:o="urn:schemas-microsoft-com:office:office"xmlns:w="urn:schemas-microsoft-com:office:word"xmlns="http://www.w3.org/TR/REC-html40"><head><meta name=Title content="TCS: estimating gene genealogies"><meta name=Keywords content=""><meta http-equiv=Content-Type content="text/html; charset=macintosh"><meta name=ProgId content=Word.Document><meta name=Generator content="Microsoft Word 11"><meta name=Originator content="Microsoft Word 11"><link rel=File-List href="TCS1.21_files/filelist.xml"><link rel=Edit-Time-Data href="TCS1.21_files/editdata.mso"><!--[if !mso]><style>v\:* {behavior:url(#default#VML);}o\:* {behavior:url(#default#VML);}w\:* {behavior:url(#default#VML);}.shape {behavior:url(#default#VML);}</style><![endif]--><title>TCS: estimating gene genealogies</title><!--[if gte mso 9]><xml> <o:DocumentProperties>  <o:Author>David Posada</o:Author>  <o:Template>Normal</o:Template>  <o:LastAuthor>David Posada</o:LastAuthor>  <o:Revision>2</o:Revision>  <o:LastPrinted>2005-07-01T07:06:00Z</o:LastPrinted>  <o:Created>2005-07-01T07:06:00Z</o:Created>  <o:LastSaved>2005-07-01T07:06:00Z</o:LastSaved>  <o:Pages>6</o:Pages>  <o:Words>2436</o:Words>  <o:Characters>13890</o:Characters>  <o:Company>Brigham Young University</o:Company>  <o:Lines>115</o:Lines>  <o:Paragraphs>27</o:Paragraphs>  <o:CharactersWithSpaces>17057</o:CharactersWithSpaces>  <o:Version>11.257</o:Version> </o:DocumentProperties></xml><![endif]--><!--[if gte mso 9]><xml> <w:WordDocument>  <w:ActiveWritingStyle Lang="EN-US" VendorID="5" DLLVersion="2" NLCheck="0">3</w:ActiveWritingStyle>  <w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery>  <w:DisplayVerticalDrawingGridEvery>0</w:DisplayVerticalDrawingGridEvery>  <w:UseMarginsForDrawingGridOrigin/>  <w:Compatibility>   <w:FootnoteLayoutLikeWW8/>   <w:ShapeLayoutLikeWW8/>   <w:AlignTablesRowByRow/>   <w:ForgetLastTabAlignment/>   <w:DoNotUseHTMLParagraphAutoSpacing/>   <w:LayoutRawTableWidth/>   <w:LayoutTableRowsApart/>   <w:UseWord97LineBreakingRules/>  </w:Compatibility>  <w:DocumentVariables>   <w:EN_Doc_Font_List_Name>&#1;&#1;&#5;Times</w:EN_Doc_Font_List_Name>   <w:EN_Lib_Name_List_Name>05Total</w:EN_Lib_Name_List_Name>   <w:EN_Main_Body_Style_Name>Mol. Biol. Evol.</w:EN_Main_Body_Style_Name>  </w:DocumentVariables> </w:WordDocument></xml><![endif]--><style><!-- /* Font Definitions */@font-face	{font-family:"Times New Roman";	panose-1:0 2 2 6 3 5 4 5 2 3;	mso-font-charset:0;	mso-generic-font-family:auto;	mso-font-pitch:variable;	mso-font-signature:50331648 0 0 0 1 0;}@font-face	{font-family:"Courier New";	panose-1:0 2 7 3 9 2 2 5 2 4;	mso-font-charset:0;	mso-generic-font-family:auto;	mso-font-pitch:variable;	mso-font-signature:50331648 0 0 0 1 0;}@font-face	{font-family:Geneva;	panose-1:0 2 11 5 3 3 4 4 4 2;	mso-font-charset:0;	mso-generic-font-family:auto;	mso-font-pitch:variable;	mso-font-signature:50331648 0 0 0 1 0;}@font-face	{font-family:Wingdings;	panose-1:0 5 2 1 2 1 8 4 8 7;	mso-font-charset:2;	mso-generic-font-family:auto;	mso-font-pitch:variable;	mso-font-signature:0 0 256 0 -2147483648 0;}@font-face	{font-family:Optima;	panose-1:0 2 0 5 3 6 0 0 2 0;	mso-font-charset:0;	mso-generic-font-family:auto;	mso-font-pitch:variable;	mso-font-signature:50331648 0 0 0 1 0;}@font-face	{font-family:Monaco;	panose-1:0 2 0 5 0 0 0 0 0 0;	mso-font-charset:0;	mso-generic-font-family:auto;	mso-font-pitch:variable;	mso-font-signature:50331648 0 0 0 1 0;} /* Style Definitions */p.MsoNormal, li.MsoNormal, div.MsoNormal	{mso-style-parent:"";	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	font-size:12.0pt;	font-family:Times;}h1	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:1;	font-size:12.0pt;	font-family:Geneva;	color:black;	mso-font-kerning:0pt;}h2	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	line-height:200%;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:2;	font-size:12.0pt;	font-family:Times;}h3	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:3;	font-size:11.0pt;	font-family:Times;}h4	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:4;	font-size:11.0pt;	font-family:Times;	color:black;	font-weight:normal;	font-style:italic;}h5	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:5;	font-size:11.0pt;	font-family:Times;	color:black;	text-decoration:underline;	text-underline:single;}h6	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:6;	font-size:11.0pt;	font-family:Times;	color:black;}p.MsoHeading7, li.MsoHeading7, div.MsoHeading7	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	text-align:center;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:7;	font-size:18.0pt;	font-family:"Times New Roman";	font-weight:bold;}p.MsoHeading8, li.MsoHeading8, div.MsoHeading8	{mso-style-next:Normal;	margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	page-break-after:avoid;	mso-outline-level:8;	font-size:12.0pt;	font-family:"Times New Roman";	color:black;	font-style:italic;}p.MsoFooter, li.MsoFooter, div.MsoFooter	{margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	tab-stops:center 216.0pt right 432.0pt;	font-size:12.0pt;	font-family:Times;}p.MsoBodyText, li.MsoBodyText, div.MsoBodyText	{margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	font-size:10.0pt;	font-family:Geneva;}p.MsoBodyTextIndent, li.MsoBodyTextIndent, div.MsoBodyTextIndent	{margin-top:0cm;	margin-right:0cm;	margin-bottom:0cm;	margin-left:36.0pt;	margin-bottom:.0001pt;	text-indent:-36.0pt;	mso-pagination:widow-orphan;	font-size:12.0pt;	font-family:"Times New Roman";}p.MsoBodyText2, li.MsoBodyText2, div.MsoBodyText2	{margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	font-size:11.0pt;	font-family:Times;	color:black;}p.MsoBodyText3, li.MsoBodyText3, div.MsoBodyText3	{margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	font-size:13.0pt;	font-family:Times;	color:black;	font-style:italic;}p.MsoBodyTextIndent2, li.MsoBodyTextIndent2, div.MsoBodyTextIndent2	{margin-top:0cm;	margin-right:0cm;	margin-bottom:0cm;	margin-left:36.0pt;	margin-bottom:.0001pt;	text-indent:-36.0pt;	mso-pagination:widow-orphan;	font-size:12.0pt;	font-family:"Times New Roman";	color:black;}a:link, span.MsoHyperlink	{color:blue;	text-decoration:underline;	text-underline:single;}a:visited, span.MsoHyperlinkFollowed	{color:purple;	text-decoration:underline;	text-underline:single;}p.MsoPlainText, li.MsoPlainText, div.MsoPlainText	{margin:0cm;	margin-bottom:.0001pt;	mso-pagination:widow-orphan;	font-size:10.0pt;	font-family:"Courier New";}table.MsoNormalTable	{mso-style-parent:"";	font-size:10.0pt;	font-family:Times;}@page Section1	{size:612.0pt 792.0pt;	margin:72.0pt 90.0pt 72.0pt 90.0pt;	mso-header-margin:36.0pt;	mso-footer-margin:36.0pt;	mso-even-footer:url(":TCS1.21_files:header.htm") ef1;	mso-footer:url(":TCS1.21_files:header.htm") f1;	mso-paper-source:0;}div.Section1	{page:Section1;} /* List Definitions */@list l0	{mso-list-id:1;	mso-list-type:simple;	mso-list-template-ids:984073;}@list l0:level1	{mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;}@list l1	{mso-list-id:2;	mso-list-type:simple;	mso-list-template-ids:984073;}@list l1:level1	{mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;}@list l2	{mso-list-id:3;	mso-list-type:simple;	mso-list-template-ids:984073;}@list l2:level1	{mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;}@list l3	{mso-list-id:4;	mso-list-type:simple;	mso-list-template-ids:0;}@list l3:level1	{mso-level-start-at:5;	mso-level-number-format:bullet;	mso-level-text:-;	mso-level-tab-stop:36.0pt;	mso-level-number-position:left;	text-indent:-18.0pt;	font-family:"Times New Roman";}@list l4	{mso-list-id:5;	mso-list-type:simple;	mso-list-template-ids:0;}@list l4:level1	{mso-level-start-at:5;	mso-level-text:%1;	mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;}@list l5	{mso-list-id:6;	mso-list-type:simple;	mso-list-template-ids:0;}@list l5:level1	{mso-level-number-format:bullet;	mso-level-text:-;	mso-level-tab-stop:54.0pt;	mso-level-number-position:left;	margin-left:54.0pt;	text-indent:-18.0pt;	font-family:"Times New Roman";}@list l6	{mso-list-id:7;	mso-list-type:simple;	mso-list-template-ids:0;}@list l6:level1	{mso-level-number-format:bullet;	mso-level-text:-;	mso-level-tab-stop:54.0pt;	mso-level-number-position:left;	margin-left:54.0pt;	text-indent:-18.0pt;	font-family:"Times New Roman";}@list l7	{mso-list-id:8;	mso-list-type:simple;	mso-list-template-ids:0;}@list l7:level1	{mso-level-start-at:5;	mso-level-text:%1;	mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;}@list l8	{mso-list-id:9;	mso-list-type:simple;	mso-list-template-ids:0;}@list l8:level1	{mso-level-number-format:bullet;	mso-level-text:-;	mso-level-tab-stop:54.0pt;	mso-level-number-position:left;	margin-left:54.0pt;	text-indent:-18.0pt;	font-family:"Times New Roman";}@list l9	{mso-list-id:10;	mso-list-type:simple;	mso-list-template-ids:0;}@list l9:level1	{mso-level-start-at:5;	mso-level-text:%1;	mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;}@list l10	{mso-list-id:11;	mso-list-type:simple;	mso-list-template-ids:0;}@list l10:level1	{mso-level-start-at:6;	mso-level-number-format:bullet;	mso-level-text:-;	mso-level-tab-stop:18.0pt;	mso-level-number-position:left;	margin-left:18.0pt;	text-indent:-18.0pt;	font-family:"Times New Roman";}@list l11	{mso-list-id:1659266496;	mso-list-type:hybrid;	mso-list-template-ids:-975133858 -1417383112 197641 328713 66569 197641 328713 66569 197641 328713;}@list l11:level1	{mso-level-start-at:2;	mso-level-number-format:bullet;	mso-level-text:-;	mso-level-tab-stop:36.0pt;	mso-level-number-position:left;	text-indent:-18.0pt;	font-family:"Times New Roman";}ol	{margin-bottom:0cm;}ul	{margin-bottom:0cm;}--></style><!--[if gte mso 9]><xml> <o:shapedefaults v:ext="edit" spidmax="2050"/></xml><![endif]--><!--[if gte mso 9]><xml> <o:shapelayout v:ext="edit">  <o:idmap v:ext="edit" data="1"/> </o:shapelayout></xml><![endif]--></head><body bgcolor=white lang=EN-US link=blue vlink=purple style='tab-interval:36.0pt'><div class=Section1><p class=MsoHeading7><span style='font-size:16.0pt;font-family:Optima'>TCS 1.21(30 June 2005)<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoBodyText2 align=center style='text-align:center'><spanstyle='font-size:10.0pt'><!--[if gte vml 1]><v:shapetype id="_x0000_t75" coordsize="21600,21600" o:spt="75" o:preferrelative="t" path="m@4@5l@4@11@9@11@9@5xe" filled="f" stroked="f"> <v:stroke joinstyle="miter"/> <v:formulas>  <v:f eqn="if lineDrawn pixelLineWidth 0"/>  <v:f eqn="sum @0 1 0"/>  <v:f eqn="sum 0 0 @1"/>  <v:f eqn="prod @2 1 2"/>  <v:f eqn="prod @3 21600 pixelWidth"/>  <v:f eqn="prod @3 21600 pixelHeight"/>  <v:f eqn="sum @0 0 1"/>  <v:f eqn="prod @6 1 2"/>  <v:f eqn="prod @7 21600 pixelWidth"/>  <v:f eqn="sum @8 21600 0"/>  <v:f eqn="prod @7 21600 pixelHeight"/>  <v:f eqn="sum @10 21600 0"/> </v:formulas> <v:path o:extrusionok="f" gradientshapeok="t" o:connecttype="rect"/> <o:lock v:ext="edit" aspectratio="t"/></v:shapetype><v:shape id="_x0000_i1025" type="#_x0000_t75" style='width:58pt; height:58pt' o:allowoverlap="f"> <v:imagedata src="TCS1.21_files/image001.png" o:title="TCSicon"/></v:shape><![endif]--><![if !vml]><img width=58 height=58src="TCS1.21_files/image002.gif" v:shapes="_x0000_i1025"><![endif]></span><spanstyle='font-size:10.0pt;font-family:Optima'><o:p></o:p></span></p><p class=MsoBodyText2 align=center style='text-align:center'><spanstyle='font-size:10.0pt;font-family:Optima'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoBodyText2 align=center style='text-align:center'><spanstyle='font-size:10.0pt;font-family:Optima'><b>2000-2005 © Mark Clement, JacobDerington (Brigham Young University, USA)<o:p></o:p></b></span></p><p class=MsoBodyText2 align=center style='text-align:center'><spanstyle='font-size:10.0pt;font-family:Optima'><b>Steve Woolley (<ahref="http://www.google.com/search?num=100&amp;hl=en&amp;client=safari&amp;rls=en&amp;q=Washington+University&amp;spell=1"></a>WashingtonUniversity, USA) and David Posada (University of Vigo, Spain).<o:p></o:p></b></span></p><p class=MsoBodyText2><span style='font-size:10.0pt;font-family:Optima'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal align=center style='margin-left:36.0pt;text-align:center;text-indent:-36.0pt'><span style='font-size:10.0pt;font-family:Optima'><i><ahref="mailto:dposada@uvigo.es">dposada@uvigo.es</a><o:p></o:p></i></span></p><p class=MsoNormal align=center style='text-align:center'><spanstyle='font-size:10.0pt;font-family:Optima'><i><ahref="http://darwin.uvigo.es/software/tcs.html">http://darwin.uvigo.es/software/tcs.html</a><o:p></o:p></i></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoPlainText><span style='font-family:Optima'><b>DISCLAIMER<o:p></o:p></b></span></p><div style='border:none;border-bottom:solid windowtext .5pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoPlainText style='border:none;mso-border-bottom-alt:solid windowtext .5pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-family:Optima'><i>Thisprogram is free software; you can redistribute it and/or modify it under theterms of the GNU General Public License as published by the Free SoftwareFoundation; either version 2 of the License, or (at your option) any laterversion. This program is distributed in the hope that it will be useful, butWITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY orFITNESS FOR A PARTICULAR PURPOSE.<span style="mso-spacerun: yes">&nbsp;</span>See the GNU General Public License for more details. You should havereceived a copy of the GNU General Public License along with this program; ifnot, write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,Boston, MA<span style="mso-spacerun: yes">&nbsp; </span>02111-1307, USA.<o:p></o:p></i></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal style='text-align:justify'><span style='font-size:10.0pt;font-family:Optima'><b>HISTORY</b></span><span style='font-size:10.0pt;font-family:Optima;color:black'><o:p></o:p></span></p><p class=MsoBodyTextIndent2><span style='font-size:10.0pt;font-family:Optima'>Version1.21 (30 June 2005) Fixed the mapping code to correctly deal with gaps asdefined in the GUI, either as 5th state or as missing (IUPAC ambiguity charactersare treated as missing data)<o:p></o:p></span></p><p class=MsoBodyTextIndent2><span style='font-size:10.0pt;font-family:Optima'>Version1.20 (25 June 2005). Fixed &quot;Save Graph&quot;. GUI behavior improvement.Added code to map character substitutions to the branches. Aesthetic changes.IUPAC warning will appear once<o:p></o:p></span></p><p class=MsoBodyTextIndent2><span style='font-size:10.0pt;font-family:Optima'>Version1.19 (23 April 2005): Fixed potential printing bug at TextOutputStream.java. Itshould not have affected any result.<o:p></o:p></span></p><p class=MsoBodyTextIndent2><span style='font-size:10.0pt;font-family:Optima'>Version1.18 (June 2004): Fixed gapmode, the program was always ignoring gaps (thanksto Manel Vera). Some code reorganization. Fixed a bug that prevented openinggraph files.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Version1.17 (May 2004): Fixed bug that prevented PICT or PS output. Small codereorganization <o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>Version 1-14-1.16 (May2004): (many fixes were done since the last distributed version; completedetails are given at the beginning of the file <i>dna.java</i></span><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>) Fixed a bug thatresulted in incorrect connections in some special cases. Improved the PICToutput format. Removed the nesting option, which was available by mistake.Allow user the select the confidence level for the parsimony limit. Addedoption to automatically select the root (assumes root is the rectangular node).Improved GUI. The program can read IUPAC symbols and will treat them as missingdata. Fixed other minor bugs<o:p></o:p></span></p><p class=MsoBodyTextIndent2><span style='font-size:10.0pt;font-family:Optima'>Version1.13: Fixed bug that was creating several unconnected haplotypes (when theyshould be connected). Maybe the same bug we thought we fixed in version 1.12.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>Version 1.07-1.12:Fixed bug that was creating several unconnected haplotypes (when they should beconnected) for some big data sets. The progress of the calculations are showedin the GUI.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>Version 1.06: severalcosmetic changes and some bugs fixed<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>Version 1.02: outgroupweights estimation included<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>Version 1.01: distancesfile included <o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>Version 1.00: Firstversion of the program. <o:p></o:p></span></p><h1><span style='font-size:10.0pt;font-family:Optima'>PROGRAM CITATION<o:p></o:p></span></h1><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoBodyText style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-family:Optima'>Clement,M., D. Posada and K. A. Crandall 2000. TCS: a computer program to estimate genegenealogies. Molecular Ecology 9 (10): 1657-1660.<o:p></o:p></span></p><p class=MsoBodyText style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><h6><span style='font-size:10.0pt;font-family:Optima'>INSTALLATION<o:p></o:p></span></h6><h2 style='line-height:normal'><span style='font-size:10.0pt;font-family:Optima'>Java<o:p></o:p></span></h2><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>First ofall, make sure that you have a Java Virtual Machine (JVM) properly installed inyour system. To test your JVM<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>1) Go to <ahref="http://javatester.org/version.html">http://javatester.org/version.html</a><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>2)Or in a terminal window, type “java &#8211;version”.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>TheJVM is included also in:<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:l11 level1 lfo12;tab-stops:list 36.0pt'><![if !supportLists]><span style='font-size:10.0pt;font-family:"Times New Roman";color:black'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span><![endif]><span style='font-size:10.0pt;font-family:Optima;color:black'>Java Runtime Environment (JRE)<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-18.0pt;mso-list:l11 level1 lfo12;tab-stops:list 36.0pt'><![if !supportLists]><span style='font-size:10.0pt;font-family:"Times New Roman";color:black'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span><![endif]><span style='font-size:10.0pt;font-family:Optima;color:black'>Java 2 Platform Standard Edition (J2SE)<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Moreinformation on obtaining the JVM in: <a href="http://java.sun.com/"title="Main Java site">http://java.sun.com/</a><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Toautomatically download the JVM: <ahref="http://java.sun.com/webapps/getjava/BrowserRedirect"title="Get JVM automatically">http://java.sun.com/webapps/getjava/BrowserRedirect</a><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Windows:The latest version of Java, 1.4.2 works fine<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Unix-like:The latest version of Java, 1.4.2 should work fine<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>MacOSX 10.4.1: The latest versions of Java, 1.4.2_07 or 1.5.0_02 work fine. <o:p></o:p></span></p><h1><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h1><h1><span style='font-size:10.0pt;font-family:Optima'>Starting TCS<o:p></o:p></span></h1><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>AfterJava is properly installed, to run TCS you should not have to do nothing otherthan decompress the compressed distribution files. Just use the executablesfile in the <i>bin</i></span><span style='font-size:10.0pt;font-family:Optima;color:black'> folder. Do not change the location of the different files withinthe program folder. <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Suggestion(from Christoph Held) for Windows:<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>1)Create a directory somewhere you like and name it e.g. &quot;TCS&quot;.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>2)Unzip the content of the file you have downloaded to this directory <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>(tickthe option &quot;use folder names&quot; or similar in your zip program to <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>maintainthe directory structure of TCS. No further installation steps <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>areneeded.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>3)Inside the directory you have created you will find a &quot;TCS1.19&quot; <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>sub-directory.To get TCS to run, double-click on the file &quot;TCS1.19.jar&quot; <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>insidethis subdirectory. (The numbers in the pathname will probably <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>changein future versions, but you get the idea.)<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>4)If all that happens is that a zip program window popping up, you will <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>haveto deactivate the association of the &quot;*.jar&quot; filetype with your ZIP <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Programin its options tab.<o:p></o:p></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><h1><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h1><h1><span style='font-size:10.0pt;font-family:Optima'>BACKGROUND<o:p></o:p></span></h1><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></i></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i>TCS</i></span><span style='font-size:10.0pt;font-family:Optima;color:black'>is acomputer program that implements the estimation of gene genealogies from DNAsequences as described by </span><!--[if supportFields]><span style='font-size:10.0pt;font-family:Optima;color:black'><span style='mso-element:field-begin'></span><spanstyle="mso-spacerun: yes">&nbsp;</span>ADDIN ENRfu <span style='mso-element:field-separator'></span></span><![endif]--><span style='font-size:10.0pt;font-family:Optima;color:black'>(Templeton et al. 1992)<!--[if gte mso 9]><xml> <w:data>440034003200440033004500390045003000300037004500410033003600380035003600310030003000300030003800330032004300460036003500360045003700320036003600300030004200350033003300390030003200300030003000300038004600380034003700300032003000300030003000300030003100370030004500440043003300300042004300300032003000300030003000300030003100380035004200350034003600350036004400370030003600430036003500370034003600460036004500320043003200300033003100330039003300390033003200320030003200330033003200330034003300310033003100350044003000300030004200300030003000300030003000</w:data></xml><![endif]--></span><!--[if supportFields]><span style='font-size:10.0pt;font-family:Optima;color:black'><span style='mso-element:field-end'></span></span><![endif]--><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>. This cladogramestimation method is also known as statistical parsimony. Some usefulreferences are indicated below.<b><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b>Limitsof parsimony (estimated/user defined)</b></span><span style='font-size:10.0pt;font-family:Optima'><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>Theprobability of parsimony (as defined in Templeton <i>et al.</i></span><spanstyle='font-size:10.0pt;font-family:Optima'> [1992], equations 6, 7, and 8) iscalculated for DNA pairwise differences until the probability exceeds, bydefault, 0.95.<span style="mso-spacerun: yes">&nbsp; </span>The number ofmutational differences associated with the probability just before this 95%cutoff is then the maximum number of mutational connections between pairs of sequencesjustified by the &quot;parsimony&quot; criterion. The user can set up adifferent cutoff, from 90% to 99%. Alternatively, the exact limit (i.e., thenumber of differences) can be set by the user (see Figure 1).<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><h4><span style='font-size:10.0pt;font-family:Optima'><b>Note: TCS is not forRFLPs<o:p></o:p></b></span></h4><p class=MsoBodyText2><span style='font-size:10.0pt;font-family:Optima'>TCScalculations for the probability of parsimony are only for DNA sequence data.If your data is RFLPs you might think you could input absolute distances, butthat would not work. The problem is that for each pair of RFLP haplotypes, theparsimony connection limit could be different, depending on the number of sharedsites. This is because for RFLPs the total number of characters minus thenumber of characters with a different state does not necessarily equal thenumber of shared characters (which is true for DNA sequences). The differencewith DNA sequences is that ++ is a shared site, while -- is not a shared site.But you could build an RFLP network by hand. <o:p></o:p></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><b>PROGRAMFILES</b></span><span style='font-size:10.0pt;font-family:Optima'><o:p></o:p></span></p><h2 style='line-height:normal'><span style='font-size:10.0pt;font-family:Optima'>Inputfiles<o:p></o:p></span></h2><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>The TCSsoftware works with aligned nucleotide sequence. It opens DNA alignment filesin either Nexus [Maddison, 1997 #2791] or PHYLIP </span><!--[if supportFields]><spanstyle='font-size:10.0pt;font-family:Optima'><span style='mso-element:field-begin'></span><spanstyle="mso-spacerun: yes">&nbsp;</span>ADDIN ENRfu <span style='mso-element:field-separator'></span></span><![endif]--><span style='font-size:10.0pt;font-family:Optima'>(Felsenstein 1991)<!--[if gte mso 9]><xml> <w:data>380042004100410030003700460037003000300037004500410033003600380035003600310030003000300030003800330032004300460036003500360045003700320036003600300030004200340042003300430037004500300030003000300036003000410032003800300032003000300030003000300030003100320030003000370037004100310043003000300032003000300030003000300030003100380035004200340036003600350036004300370033003600350036004500370033003700340036003500360039003600450032004300320030003300310033003900330039003300310032003000320033003300380033003200350044003000300030003000300030003000300030003000</w:data></xml><![endif]--></span><!--[if supportFields]><span style='font-size:10.0pt;font-family:Optima'><span style='mso-element:field-end'></span></span><![endif]--><spanstyle='font-size:10.0pt;font-family:Optima'> <b>sequential</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> format.<span style="mso-spacerun:yes">&nbsp; </span>Alternatively, absolute distance files in <i>modified</i></span><spanstyle='font-size:10.0pt;font-family:Optima'> NEXUS or PHYLIP files can also beused.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>Sequencesdo not need to be collapsed into haplotypes, as frequency data can beincorporated into the output.<span style="mso-spacerun: yes">&nbsp; </span>Theprogram collapses sequences into haplotypes and calculates the frequencies ofthe haplotypes in the sample.<span style="mso-spacerun: yes">&nbsp;</span>These frequencies are used to estimate haplotype outgroup probabilities,which correlate with haplotype age </span><!--[if supportFields]><spanstyle='font-size:10.0pt;font-family:Optima'><span style='mso-element:field-begin'></span><spanstyle="mso-spacerun: yes">&nbsp;</span>ADDIN ENRfu <span style='mso-element:field-separator'></span></span><![endif]--><span style='font-size:10.0pt;font-family:Optima'>(Donnelly and Tavaré 1986; Castelloe and Templeton 1994)<!--[if gte mso 9]><xml> <w:data>3800370030003300380031003800330030003000410036004100330036003800350036003100300030003000300038003300320043004600360035003600450037003200360036003000300042003500320032003200310036003400300030003100330032003700430045003000320030003000300030003000300033003800440046004500450034003600370036003000320030003000300030003000300032004300350042003400330036003100370033003700340036003500360043003600430036004600360035003200430032003000330031003300390033003900330034003200300032003300330031003300360033003900330042003200300034003400360046003600450036004500360035003600430036004300370039003200430032003000330031003300390033003800330036003200300032003300330033003300380033003900350044003000300030004300300030003000300030003000</w:data></xml><![endif]--></span><!--[if supportFields]><span style='font-size:10.0pt;font-family:Optima'><span style='mso-element:field-end'></span></span><![endif]--><spanstyle='font-size:10.0pt;font-family:Optima'>. <span style='color:black'><b><o:p></o:p></b></span></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Someexamples:<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><h4><span style='font-size:10.0pt;font-family:Optima;font-style:normal'><u>AlignedDNA sequences<o:p></o:p></u></span></h4><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></i></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i>Thisis sequential NEXUS:<o:p></o:p></i></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>#NEXUS<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Begindata;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Dimensionsntax=4 nchar=6;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Formatdatatype=nucleotide gap=- missing=? ;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Matrix<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq1<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>AAAAA-<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq2<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>AAAAC-<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq3<spanstyle="mso-spacerun: yes">&nbsp;&nbsp; </span><span style="mso-spacerun:yes">&nbsp;&nbsp;&nbsp;</span>AAAAA?<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq4<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>AAAAAA<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>End;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i>andthis is sequential Phylip:<o:p></o:p></i></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>46<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq1AAAAA-<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq2AAAAC-<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq3AAAAA?<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq4AAAAAA<o:p></o:p></span></p><h5><span style='font-size:10.0pt;font-family:Optima;font-weight:normal'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h5><h5><span style='font-size:10.0pt;font-family:Optima;font-weight:normal'>Distancefile<o:p></o:p></span></h5><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Anoption exists to read a matrix of absolute distances among HAPLOTYPES. Thematrix should be LOWER DIAGONAL in NEXUS (example_dis.nex) or PHYLIP(example_dis.phy) format.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>IMPORTANT:you have to add the &quot;nchar&quot; to these files, so the 95% connectionlimit can be calculated. Look a the example files:<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><h4><span style='font-size:10.0pt;font-family:Optima'>This is a modified NEXUSformat:<o:p></o:p></span></h4><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>#NEXUS<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Begintaxa;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Dimensionsntax=10;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Taxlabels<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq4<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq5<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq6<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq7<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq8<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq9<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Seq10<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>End;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Begindistances;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Formattriangle=lower labels nodiagonal;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Matrix<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq1<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq2<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq3<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq4<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq5<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4<spanstyle="mso-spacerun: yes">&nbsp; </span>4<span style="mso-spacerun: yes">&nbsp;</span>4<span style="mso-spacerun: yes">&nbsp; </span>3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq6<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4<spanstyle="mso-spacerun: yes">&nbsp; </span>4 <span style="mso-spacerun:yes">&nbsp;</span>4<span style="mso-spacerun: yes">&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq7<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>1<span style="mso-spacerun: yes">&nbsp;</span>1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq8<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4<spanstyle="mso-spacerun: yes">&nbsp; </span>4<span style="mso-spacerun: yes">&nbsp;</span>4<span style="mso-spacerun: yes">&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq9<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq10<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>1<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>1<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><spanstyle='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>;<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>End;<b><o:p></o:p></b></span></p><h4><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h4><h4><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h4><h4><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h4><h4><span style='font-size:10.0pt;font-family:Optima'>and this is a modifiedPHYLIP format<o:p></o:p></span></h4><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>10404<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq1<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq2<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq3<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq4<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq5<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>4<span style="mso-spacerun: yes">&nbsp; </span>4<spanstyle="mso-spacerun: yes">&nbsp; </span>4<span style="mso-spacerun: yes">&nbsp;</span>3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq6<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>4<span style="mso-spacerun: yes">&nbsp; </span>4<spanstyle="mso-spacerun: yes">&nbsp; </span>4<span style="mso-spacerun: yes">&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>2<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq7<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>1<spanstyle="mso-spacerun: yes">&nbsp; </span>1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq8<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>4<span style="mso-spacerun: yes">&nbsp; </span>4<spanstyle="mso-spacerun: yes">&nbsp; </span>4<span style="mso-spacerun: yes">&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq9<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span>3<span style="mso-spacerun: yes">&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>3<spanstyle="mso-spacerun: yes">&nbsp; </span>3<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>3<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:8.0pt;font-family:Monaco;color:black'>Seq10<spanstyle="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>1<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>2<span style="mso-spacerun: yes">&nbsp; </span>1<spanstyle="mso-spacerun: yes">&nbsp; </span>2<span style="mso-spacerun: yes">&nbsp;</span>1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><h3><span style='font-size:10.0pt;font-family:Optima'>Logfile<o:p></o:p></span></h3><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>Each timethat the TCS analysis is performed, a log file is saved (<i>*.log</i></span><spanstyle='font-size:10.0pt;font-family:Optima'>). This file contains informationon the run: probabilities of parsimony for mutational steps, the pairwiseabsolute distance matrix, a test listing of connections made and missingintermediates generated, outgroup weights for each haplotype, a graphdescription, and the date and time elapsed for the analysis.<spanstyle='color:black'><o:p></o:p></span></span></p><h3><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h3><h3><span style='font-size:10.0pt;font-family:Optima'>Graph file<o:p></o:p></span></h3><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima'>Each time that the TCS analysis is performed, a graph file(GML format) is saved. The name of this file will be <i>*.graph.</i></span><spanstyle='font-size:10.0pt;font-family:Optima'> This graph can be opened later inTCS, where it can be modified and saved again.<o:p></o:p></span></p><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><h3><span style='font-size:10.0pt;font-family:Optima'>RUNNING TCS<o:p></o:p></span></h3><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>1. Openthe DNA data file in the File menu<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>2. Clickon RUN<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>3. Theprogram reads the file and collapses sequences to haplotypes<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>4. Anabsolute distance matrix is then calculated for all pairwise comparisons ofhaplotypes. <o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>5. Theparsimony connection limit is calculated. Alternatively, this limit can be setup by the user (see Figure 1).<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'>6. Thesejustified connections are then made resulting in a (by default) 95% set ofplausible networks (1 or more)<span style="mso-spacerun: yes">&nbsp; </span><o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>7. A graph is generated and automatically opened. In thisgraph, haplotypes are drawn in a size proportional to their frequency.<o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'><b>Showing changes<o:p></o:p></b></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>By clicking on the button “show changes” the program willdisplay changes across branches. Ambiguous assignations will show an asterisk(*) before the site number. <span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>The exact location of a change along branches of length&gt; 1 is at random.<o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><h3 style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>Editingthe graph<o:p></o:p></span></h3><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Youcan select (by clicking), create and delete nodes (haplotypes) o branches onthe graph. Automatic algorithms to order the graph are available in the menus.You can move the nodes and branches around and save the file as GML (thisformat will be recognized by TCS later, if you want to edit further the graph)or as postscript or PICT file. By double-clicking on a haplotype node, you willbe able of displaying its frequency and its outgroup weight. The haplotype in asquare has the biggest outgroup weight.</span><span style='font-size:10.0pt;font-family:Optima'><o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><h3 style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>Printing<o:p></o:p></span></h3><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>The graph is printed by being saved as a postscript fileand sent manually to the printer or as a PICT file. In MacOS X the Grab toolcan be easily used to obtain a TIFF file of the corresponding portion of thescreen.<o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><h3 style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>Executiontimes<o:p></o:p></span></h3><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>The program can handle a reasonable number ofsequences.<span style="mso-spacerun: yes">&nbsp; </span>For example, an HTLVdata set with 69 haplotypes of length 725 bps took over one hour to run in aMacintosh G3.<span style="mso-spacerun: yes">&nbsp; </span>Memory requirementsare low, and the program will run with less than 1 MB RAM. <o:p></o:p></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><!--[if gte vml 1]><v:shape id="_x0000_i1026" type="#_x0000_t75" style='width:409pt;height:300pt'> <v:imagedata src="TCS1.21_files/image003.png" o:title="a"/></v:shape><![endif]--><![if !vml]><img border=0 width=409 height=300src="TCS1.21_files/image004.gif" v:shapes="_x0000_i1026"><![endif]><o:p></o:p></b></span></p><p class=MsoHeading8><span style='font-size:10.0pt;font-family:Optima'>Figure1. The TCS interface<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><!--[if gte vml 1]><v:shape id="_x0000_i1027" type="#_x0000_t75" style='width:202pt;height:272pt'> <v:imagedata src="TCS1.21_files/image005.png" o:title="node9"/></v:shape><![endif]--><![if !vml]><img border=0 width=202 height=272src="TCS1.21_files/image006.jpg" v:shapes="_x0000_i1027"><![endif]><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i>Figure2. Node information. This is the information displayed when double-clicking onnode “Seq10” in Figure 1.<o:p></o:p></i></span></p><span style='font-size:10.0pt;font-family:Optima;color:black;mso-ansi-language:EN-US;mso-fareast-language:EN-US'><b><br clear=ALL style='page-break-before:always'></b></span><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b>CAVEATS<o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Thereare some things that the user of TCS needs to be aware of:<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b>Treatmentof Gaps (5<sup>th</sup> state / missing data)</b></span><span style='font-size:10.0pt;font-family:Optima;color:black'><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Bydefault, gaps are counted as events (i.e. treated as a fifth state). You canturn off this option in the program interface (Figure 1) so gaps are treated asmissing data. <o:p></o:p></span></p><h6><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></h6><h6><span style='font-size:10.0pt;font-family:Optima'>Potential problems withmissing or ambiguous data<o:p></o:p></span></h6><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Whencollapsing sequences to haplotypes, missing data may create some problems whenthe sequence <i>only differ at missing or ambiguous characters</i></span><spanstyle='font-size:10.0pt;font-family:Optima;color:black'>. Missing data maycreate some paradoxes ins such cases, and the order of the sequences may changethe results of the collapsing.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>1TGGA?AAAAAAACT<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>2TGGAAAAAAAAACT<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>3TGGACAAAAAAACT<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Itis not easy to decide whether we have 2 or 3 haplotypes. Moreover, in this dataset, TCS will say that there is 1 haplotype ... why?... well, the way TCS worksis by comparing each pair in order<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>1-2= 0<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>1-3= 0<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>therefore,there is just 1 haplotype with a frequency of 3. However if we change the orderof the sequences:<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>2TGGAAAAAAAAACT<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>1TGGA?AAAAAAACT<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>3TGGACAAAAAAACT<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>andcompare again each pair in order:<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>2-1= 0<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>2-3= 1<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Therefore,there are two haplotypes, one with a frequency of two (=2+1) and the other withfrequency one (=3). Given the length of the sequences that people is using today,this situation will be really uncommon. Anyway TCS should warn you in suchcases.<o:p></o:p></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><h6><span style='font-size:10.0pt;font-family:Optima'>Output graphs<o:p></o:p></span></h6><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'>Beaware, if you have several unconnected subnetworks, TCS will not spread thoseautomatically. If you have overlapping haplotypes, you have to move then aroundusing the mouse. Nothing should overlap.<o:p></o:p></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima'><b>Credits<o:p></o:p></b></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>Many thanks for many users reporting potential bugs andproviding suggestions.<o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal style='tab-stops:0cm'><span style='font-size:10.0pt;font-family:Optima'>For graphic purposes, TCS uses the freeware VGJ 1.0.3,distributed under the terms of the GNU General Public License, Version 2), ispackaged within the TCS program.<o:p></o:p></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima'><ahref="http://www.eng.auburn.edu/department/cse/research/graph_drawing/graph_drawing.html">http://www.eng.auburn.edu/department/cse/research/graph_drawing/graph_drawing.html</a><o:p></o:p></span></p><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'>TCS uses the <i>BrowserLauncher</i></span><spanstyle='font-size:10.0pt;font-family:Optima;color:black'> version 1.4b1<spanstyle="mso-spacerun: yes">&nbsp; </span>class by Eric Albert<o:p></o:p></span></p><p class=MsoNormal style='border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><span style='font-size:10.0pt;font-family:Optima;color:black'><b><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></b></span></p></div><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><b>USEFULREFERENCES</b></span><span style='font-size:10.0pt;font-family:Optima;color:black'><o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><!--[if supportFields]><spanstyle='font-size:10.0pt;font-family:Optima'><span style='mso-element:field-begin'></span><spanstyle="mso-spacerun: yes">&nbsp;</span>ADDIN ENBbu <span style='mso-element:field-separator'></span></span><![endif]--><span style='font-size:10.0pt;font-family:Optima'>Castelloe, J. and A. R. Templeton 1994. Root probabilitiesfor intraspecific gene trees under neutral coalescent theory. Mol. Phylogenet.Evol. <b>3:</b></span><span style='font-size:10.0pt;font-family:Optima'>102-113.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Clement, M., D. Posada and K. A.Crandall 2000. TCS: a computer program to estimate gene genealogies. MolecularEcology (in press)<b>:</b></span><span style='font-size:10.0pt;font-family:Optima'> <o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Crandall, K. A. 1994. Intraspecificcladogram estimation:<span style="mso-spacerun: yes">&nbsp; </span>Accuracy athigher levels of divergence. Syst. Biol. <b>43:</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> 222-235.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Crandall, K. A. 1995. Intraspecificphylogenetics: Support for dental transmission of human immunodeficiency virus.J. Virol. <b>69:</b></span><span style='font-size:10.0pt;font-family:Optima'>2351-2356.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Crandall, K. A. 1996a. Multipleinterespecies transmissions of human and simian T-cell leukemia/lymphoma virustype I sequences. Mol. Biol. Evol. <b>13:</b></span><span style='font-size:10.0pt;font-family:Optima'> 115-131.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Crandall, K. A. 1996b. Multipleinterspecies transmissions of human and simian T-cell leukemia/lymphoma virustype I sequences. Mol. Biol. Evol. <b>13:</b></span><span style='font-size:10.0pt;font-family:Optima'> 115-131.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Crandall, K. A. and A. R. Templeton1996. Applications of intraspecific phylogenetics. Pp. 81-99. <i>in</i></span><spanstyle='font-size:10.0pt;font-family:Optima'><span style="mso-spacerun:yes">&nbsp; </span>P. H. Harvey, A. J. Leigh Brown, J. Maynard Smith and S.Nee, eds. New Uses for New Phylogenies.<span style="mso-spacerun: yes">&nbsp;</span>Oxford University Press, Oxford, England. <o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Crandall, K. A., A. R. Templetonand C. F. Sing 1994. Intraspecific phylogenetics: problems and solutions. Pp.273-297. <i>in</i></span><span style='font-size:10.0pt;font-family:Optima'><spanstyle="mso-spacerun: yes">&nbsp; </span>R. W. Scotland, D. J. Siebert and D. M.Williams, eds. Models in Phylogeny Reconstruction.<span style="mso-spacerun:yes">&nbsp; </span>Clarendon Press, Oxford, England. <o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Donnelly, P. and S. Tavaré 1986.The ages of alleles and a coalescent. Adv. Appl. Prob. <b>18:</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> 1-19.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Felsenstein, J. 1991. PHYLIP:Phylogenetic Inference Package. 3.4. University of Washington, Seattle, WA.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Templeton, A. R. 1995. A cladisticanalysis of phenotypic associations with haplotypes inferred from restrictionendonuclease mapping or DNA sequencing.<span style="mso-spacerun: yes">&nbsp;</span>V. Analysis of case/control sampling designs: Alzheimer's disease andthe apoprotein E locus. Genetics <b>140:</b></span><span style='font-size:10.0pt;font-family:Optima'> 403-409.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Templeton, A. R. 1998. Nested cladeanalyses of phylogeographic data: testing hypotheses about gene flow andpopulation history. Molecular Ecology <b>7:</b></span><span style='font-size:10.0pt;font-family:Optima'> 381-397.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Templeton, A. R., E. Boerwinkle andC. F. Sing 1987. A cladistic analysis of phenotypic associations withhaplotypes inferred from restriction endonuclease mapping and DNA sequencedata. I. Basic theory and an analysis of alcohol dehydrogenase activity in <i>Drosophila</i></span><spanstyle='font-size:10.0pt;font-family:Optima'>. Genetics <b>117:</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> 343-351.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Templeton, A. R., K. A. Crandalland C. F. Sing 1992. A cladistic analysis of phenotypic associations withhaplotypes inferred from restriction endonuclease mapping and DNA sequencedata. III. Cladogram estimation. Genetics <b>132:</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> 619-633.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Templeton, A. R., E. Routman and C.A. Phillips 1995. Separating population structure from population history: acladistic analysis of the geographical distribution of mitochondrial DNAhaplotypes in the Tiger salamander, <i>Ambystoma tigrinum</i></span><spanstyle='font-size:10.0pt;font-family:Optima'>. Genetics <b>140:</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> 767-782.<o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'>Templeton, A. R. and C. F. Sing1993. A cladistic analysis of phenotypic associations with haplotypes inferredfrom restriction endonuclease mapping. IV. Nested analyses with cladogramuncertainty and recombination. Genetics <b>134:</b></span><spanstyle='font-size:10.0pt;font-family:Optima'> 659-669.<!--[if gte mso 9]><xml> <w:data>430041004600460032003800420043003000300034003200410033003600380035003600310030003000300030003800330032004300460036003500360045003600320036003200300030004200350033003300390030003200390030003000300038004600410034004400300032003000300030003000300042004400380045003900330038003000390034003700300030003000300030003000</w:data></xml><![endif]--></span><!--[if supportFields]><span style='font-size:10.0pt;font-family:Optima'><span style='mso-element:field-end'></span></span><![endif]--><spanstyle='font-size:10.0pt;font-family:Optima'><o:p></o:p></span></p><p class=MsoBodyTextIndent><span style='font-size:10.0pt;font-family:Optima'>PosadaD, Crandall KA (2001) Intraspecific gene genealogies: Trees grafting intonetworks. Trends Ecol Evol <b>16</b></span><span style='font-size:10.0pt;font-family:Optima'>:37-45<o:p></o:p></span></p><div style='border:none;border-bottom:solid windowtext .75pt;padding:0cm 0cm 1.0pt 0cm'><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt;border:none;mso-border-bottom-alt:solid windowtext .75pt;padding:0cm;mso-padding-alt:0cm 0cm 1.0pt 0cm'><spanstyle='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p></div><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'><i>David Posada<o:p></o:p></i></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i>July1,<span style="mso-spacerun: yes">&nbsp; </span>2005<o:p></o:p></i></span></p><p class=MsoNormal><span style='font-size:10.0pt;font-family:Optima;color:black'><i>dposada@uvigo.es<o:p></o:p></i></span></p><p class=MsoNormal style='margin-left:36.0pt;text-indent:-36.0pt'><spanstyle='font-size:10.0pt;font-family:Optima'><![if !supportEmptyParas]>&nbsp;<![endif]><o:p></o:p></span></p></div></body></html>